indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -398,4 +431,31 @@ private Response specSuspendOrResume(final String id, boolean suspend)
         }
     );
   }
+
+  private Response specSuspendAllOrResumeAll(boolean suspend)
+  {
+    return asLeaderWithSupervisorManager(
+        manager -> {
+          Set<String> supervisorIds = manager.getSupervisorIds();
```

Comment:
Might also want to put the whole operation under a lock so behavior is a bit more consistent

Reply:
> return a list of supervisor ids that had a state change
I think there is no need to return these ids since these `all` endpoints are idempotent operations. Just return OK would be fine. And I think users do not care about which supervisor has a state change, they just need to know they have changed all supervisors to suspened state or running state. Besides, return these ids would make the code a little complex and ugly ):